src/deployment/flavors.md

```diff
@@ -3,41 +3,159 @@ title: Creating flavors for Flutter
+You can use flavors to supply your app with different development versions.
 
-{% comment %}
-Eventually, we should actually include instructions,
-but by platform and we can then move the platform-specific
-instructions to /development/platform-integration/macos
-or ios or android, etc.
-sz: Yes, I've already asked Shamira (who will be working
-on the new docs), to plan for multiple pages.
-{% endcomment %}
-
-Do you need to set up product flavors for different development
-environments or release types?
-The community has written some articles and packages you might find useful.
-These articles address flavors for both iOS and Android.
-
-* [Creating flavors of a Flutter app][]
-* [Flavoring Flutter][]
-* [Flutter Ready to Go][]
-* [Build flavors in Flutter (Android and iOS)
-  with different Firebase projects per flavor][]
-* [Flutter 1.17 — no more Flavors, no more iOS Schemas.
-  Command argument that changes everything][]
-* [Multi-environment Flutter Projects with Flavors][]
-  (including support for web, Firebase and FlutterFire CLI)
-
-The following packages are listed alphabetically:
+Imagine that you want to make both free and paid versions of your Flutter app. 
+You can use flavors to set up both app versions 
+without writing two separate apps. 
+The free version of the app has basic functionality. 
+In contrast, the paid version has basic app functionality, extra features, 
+and different styles for paid users. 
+
+Another example where you could use flavors is for feature development. 
+If you’ve built a new feature and want to try it out, 
+you could set up a flavor and deploy that feature to a different API endpoint 
+than production. Your production code remains unaffected 
+until you are ready to deploy your new feature to production.
+
+Flavors function by letting you define compile time configurations 
+and set parameters that are read at runtime depending on your app needs.
+
+This document guides you through setting up Flutter flavors for iOS and Android. 
+
+## Environment set up
+Prerequisites:
+* Xcode installed
+* Android Studio
+* An existing project 
+
+To set up flavors in iOS, you’ll define build configurations in Xcode. 
+
+## Creating flavors in iOS
+1. Open your project in Xcode.
+
+2. Select the **Runner > New Scheme** from the menu to add a new Scheme. 
+A scheme describes how Xcode runs different actions. 
+For the purposes of this guide the example _flavor_ and _scheme_ are 
+named `development`. 
+The build configurations in the development scheme 
+have the `-development` suffix. 
+
+3. Duplicate the build configurations to differentiate between the 
+default configurations that are already available and the new configurations 
+for the `development` scheme. Under the **Info** tab at the end of the 
+**Configurations** dropdown list, click the plus button and duplicate 
+each configuration name(Debug, Release, and Profile). 
+Duplicate the listed production configurations, once for each environment.   
```

Comment:
Your instructions are clear, but people get confused at this stage.  Would be worth a note block that they should expand the configurations and make sure they are "Based on Configuration File" Debug.xcconfig or Release.xcconfig, NOT the Pods-Runner*.xcconfigs.
Just saw this in https://github.com/flutter/flutter/issues/93384#issuecomment-1331560482